lib/jekyll-mentions.rb

```diff
@@ -82,18 +93,58 @@ def mention_base(config = {})
         end
       end
 
-      # Public: Defines the conditions for a document to be emojiable.
+      # Public: Defines the conditions for a document to be mentionable.
       #
       # doc - the Jekyll::Document or Jekyll::Page
       #
       # Returns true if the doc is written & is HTML.
       def mentionable?(doc)
         (doc.is_a?(Jekyll::Page) || doc.write?) &&
-          doc.output_ext == ".html" || (doc.permalink&.end_with?("/"))
+          (doc.output_ext == ".html" || (doc.permalink&.end_with?("/"))) &&
+          !mentionable_disabled?(doc)
       end
 
       private
 
+      # Private: Returns a boolean indicating whether the frontmatter of a specific doc explicitly
+      # disables jekyll-mentions
+      #
+      # doc - the current Jekyll::Document
+      #
+      # Returns true if the frontmatter of the doc sets jekyll-mentions to false, else false
+      def mentionable_disabled?(doc)
+        page_config = page_config(doc) || {}
+        post_config = post_config(doc) || {}
+        page_config.merge(post_config)[JEKYLL_MENTIONS] == false
+      end
+
+      # Private: Finds the current post's config that's normally found in the frontmatter
+      # of the page (this method assumes that your posts are in a `_posts/` directory)
+      #
+      # doc - the current Jekyll::Document
+      #
+      # Returns a hash-like configuration of the current post's config or nil
+      def post_config(doc)
+        return unless doc.path.include?("_posts")
+
+        page_reader = Jekyll::PageReader.new(doc.site, doc.path)
+        page_reader.read([doc.path]).first.data
+      end
+
+      # Private: Finds the current page's config that's normally found in the frontmatter
+      # of the page
+      #
+      # doc - the current Jekyll::Document
+      #
+      # Returns a hash-like configuration of the current page's config or nil
+      def page_config(doc)
+        filtered_pages = doc.site.pages.select { |page| page.name == doc.path.split("/")[-1] }
```

Comment:
I'll take a look at this as well.

Reply:
I replaced this with `doc.basename + doc.extname`. This is because the `basename` doesn't inherently include the extension, but the `page.name` does.

Reply:
There is a bit of an inconsistency between `Jekyll::Page` and `Jekyll::Document` in the core.
For a `Jekyll::Page`, `basename` returns the filename without extension. For `Jekyll::Document`, `basename` returns filename **with** extension whereas `basename_without_ext` returns the filename w/o extension.

So, the better alternative here would be to use `File.basename` directly.
```ruby
filtered_pages = doc.site.pages.select { |page| page.name == File,basename(doc.path) }
```

Reply:
Ah, you're probably right. I was most likely mixing and matching them. Either way, I removed this comparison anyway. Doing string and path comparisons isn't as stable anyhow, and since this is a gem that will go from one project to another, the ability for the code to work on multiple systems and projects is important.

Just grabbing the raw data and doing a `.merge` on the hashes (no matter whether a post, draft, or page) is safer.